src/assets/Azure.Core.Shared/RetryAfterDelayStrategy.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+#nullable enable
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Implementation of a <see cref="DelayStrategy"/> of which the interval is from
+    /// retry-after header of service response.
+    /// </summary>
+    internal class RetryAfterDelayStrategy : DelayStrategy
+    {
+        protected const string RetryAfterHeaderName = "Retry-After";
+        protected const string RetryAfterMsHeaderName = "retry-after-ms";
+        protected const string XRetryAfterMsHeaderName = "x-ms-retry-after-ms";
+
+        private DelayStrategy _fallbackStrategy;
+
+        /// <summary>
+        /// Create a <see cref="RetryAfterDelayStrategy"/> with a default retry-after value which normally
+        /// comes from the initial response of an LRO operation.
+        /// </summary>
+        /// <param name="fallbackStrategy"> Fallback strategy if retry after is not present. </param>
+        public RetryAfterDelayStrategy(DelayStrategy fallbackStrategy)
+        {
+            _fallbackStrategy = fallbackStrategy;
```

Comment:
Should we check for null here just in case?

Reply:
We are the only caller and we have nullability checks on so this can't be null.  Happy to add that or even change this to a nullable param and enforce the default ConstantDelayStrategy here when its null.

Thoughts?

Reply:
Talked offline and decided to move the defaulting into the ctor here to guarantee no nulls.